### PR TITLE
Flush Clickhouse buffers faster by encoding early

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR retries https://github.com/plausible/analytics/pull/3511, but with the session insert order fixed.

I'm keeping it in draft until https://github.com/plausible/analytics/pull/3622 is merged

### Tests
- [x] Automated tests have been added https://github.com/plausible/analytics/pull/3622

### Changelog
- [x] Entry has been added to changelog

### Documentation
- [x] [Docs](https://github.com/plausible/docs) have been updated https://github.com/plausible/docs/pull/446

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: lib/plausible/session/cache_store.ex [hunk 1]
```diff
@@ -8,7 +8,7 @@ defmodule Plausible.Session.CacheStore do
     session =
       if found_session do
         updated_session = update_session(found_session, event)
-        buffer.insert([%{updated_session | sign: 1}, %{found_session | sign: -1}])
+        buffer.insert([%{found_session | sign: -1}, %{updated_session | sign: 1}])
```
Comment (ruslandoga) [inline lib/plausible/session/cache_store.ex:11]: The fix.

File: test/plausible/session/cache_store_test.exs [hunk 2]
```diff
@@ -88,7 +88,7 @@ defmodule Plausible.Session.CacheStoreTest do
 
     CacheStore.on_event(event1, nil, buffer)
     CacheStore.on_event(event2, nil, buffer)
-    assert_receive({WriteBuffer, :insert, [[session, _negative_record]]})
+    assert_receive({WriteBuffer, :insert, [[_negative_record, session]]})
```
Comment (ruslandoga) [inline test/plausible/session/cache_store_test.exs:91]: These test use a mock buffer so they had to be updated.

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +26,7 @@ All notable changes to this project will be documented in this file.
 - Show Outbound Links / File Downloads / 404 Pages / Cloaked Links instead of Goal Conversions when filtering by the corresponding goal
 - Require custom properties to be explicitly added from Site Settings > Custom Properties in order for them to show up on the dashboard
 - GA/SC sections moved to new settings: Integrations
+- `CLICKHOUSE_MAX_BUFFER_SIZE` now represents bytes and defaults to `100000` (100KB)
```
Comment (aerosol) [inline CHANGELOG.md:29]: Can you rename it to `CLICKHOUSE_MAX_BUFFER_SIZE_BYTES` so I can keep both in our deployment scripts? Makes it easier to revert and later on we can deprecate the old one.

Comment (aerosol) [reply CHANGELOG.md]: Awesome thanks!

**Review by aerosol — APPROVED** [review]

**Pull request merged**